app/selectors/login.ts

```diff
@@ -0,0 +1,22 @@
+import { createSelector } from 'reselect';
+import isEmpty from 'lodash/isEmpty';
+
+import { IApplicationState } from '../definitions';
+
+const getUser = (state: IApplicationState) => {
+	if (!isEmpty(state.share?.user)) {
+		return state.share.user;
+	}
+	return state.login?.user;
+};
+const getLoginServices = (state: IApplicationState) => state.login.services || {};
+const getShowFormLoginSetting = (state: IApplicationState) => state.settings.Accounts_ShowFormLogin || false;
```

Comment:
If you are sure of the return of this property, the right thing would be to explain its value.
As it is, it can be `number`, `string` or `boolean`, but the right thing would be to define the value, even because you already said that it can be `Accounts_ShowFormLogin` or `false`.
Something like that:
`const getShowFormLoginSetting = (state: IApplicationState) => state.settings.Accounts_ShowFormLogin as boolean || false;`

Reply:
The same for the other properties.

Reply:
Makes sense to me